### ISSUE TITLE
Recursive Detection Failing

### ISSUE THREAD

**Issue by CraigKelly** [coveooss/json-schema-for-humans#122]
I don't know if I've found a regression or an additional recursion, but I'm seeing a recursion depth exceed error.

Using a venv with Python 3.9.5:

```shell
$ python -m venv env
$ source env/bin/activate
$ python -m pip install --upgrade json-schema-for-humans
$ hash -r && pyenv rehash
$ python -m pip freeze
certifi==2021.10.8
charset-normalizer==2.0.9
click==8.0.3
dataclasses-json==0.5.6
htmlmin==0.1.12
idna==3.3
Jinja2==3.0.3
json-schema-for-humans==0.39.1
markdown2==2.4.2
MarkupSafe==2.0.1
marshmallow==3.14.1
marshmallow-enum==1.5.1
mypy-extensions==0.4.3
Pygments==2.10.0
pytz==2021.3
PyYAML==5.4.1
requests==2.26.0
typing-extensions==4.0.1
typing-inspect==0.7.1
urllib3==1.26.7
```

Then I run this command:

```shell
$ generate-schema-doc --config-file config.json ./Bug.json
```

Where `config.json` contains:

```json
{
    "link_to_reused_ref": true
}
```

And `Bug.json` contains

```json
{
    "$schema": "https://json-schema.org/draft/2020-12/schema",
    "$id": "https://schema.somewhere.com/v1.0/Bug",
    "title": "Bug",
    "description": "Display the issue.",
    "type": "object",
    "properties": {
        "Code": {
            "type": "string"
        },
        "RecursiveArray": {
            "type": "array",
            "items": {
                "$ref": "#"
            }
        },
        "DecoratedRecursiveArray": {
            "type": "array",
            "items": {
                "type": "object",
                "properties": {
                    "SomeName": {
                        "type": "string"
                    },
                    "TheThing": {
                        "$ref": "#"
                    }
                }
            }
        }
    }
}
```

The last few lines of the error are:

```
  File "/home/craig/tmp/env/lib/python3.9/site-packages/jinja2/runtime.py", line 828, in _invoke
    rv = self._func(*arguments)
  File "/home/craig/tmp/env/lib/python3.9/site-packages/json_schema_for_humans/templates/js/content.html", line 11, in template
    {%- set type_name = schema.type_name -%}
  File "/home/craig/tmp/env/lib/python3.9/site-packages/jinja2/environment.py", line 474, in getattr
    return getattr(obj, attribute)
  File "/home/craig/tmp/env/lib/python3.9/site-packages/json_schema_for_humans/schema/schema_node.py", line 415, in type_name
    name = get_type_name(self)
  File "/home/craig
[capture.txt](https://github.com/coveooss/json-schema-for-humans/files/7765655/capture.txt)
/tmp/env/lib/python3.9/site-packages/json_schema_for_humans/templating_utils.py", line 12, in get_type_name
    def _python_type_to_json_type(python_type: Type[Union[str, int, float, bool, list, dict]]) -> str:
  File "/home/craig/.pyenv/versions/3.9.5/lib/python3.9/typing.py", line 265, in inner
```

I've attached the complete output.

**Comment by CraigKelly** [comment]
[capture.txt](https://github.com/coveooss/json-schema-for-humans/files/7765658/capture.txt)

**Comment by CraigKelly** [comment]
One additional detail: changing `Bug.json` to this workaround version gets past the bug (although it is an admittedly ugly hack)...

```json
{
    "$schema": "https://json-schema.org/draft/2020-12/schema",
    "$id": "https://schema.somewhere.com/v1.0/Bug",
    "title": "Bug",
    "description": "Display the issue.",
    "type": "object",
    "properties": {
        "Code": {
            "type": "string"
        },
        "RecursiveArray": {
            "type": "array",
            "items": {
                "$ref": "#/BugWorkaround"
            }
        },
        "DecoratedRecursiveArray": {
            "type": "array",
            "items": {
                "type": "object",
                "properties": {
                    "SomeName": {
                        "type": "string"
                    },
                    "TheThing": {
                        "$ref": "#/BugWorkaround"
                    }
                }
            }
        }
    },
    "BugWorkaround": {
        "$schema": "https://json-schema.org/draft/2020-12/schema",
        "$id": "https://schema.somewhere.com/v1.0/BugWorkaround",
        "title": "BugWorkaround",
        "description": "Display the issue.",
        "type": "object",
        "properties": {
            "Code": {
                "type": "string"
            },
            "RecursiveArray": {
                "type": "array",
                "items": {
                    "$ref": "#/BugWorkaround"
                }
            },
            "DecoratedRecursiveArray": {
                "type": "array",
                "items": {
                    "type": "object",
                    "properties": {
                        "SomeName": {
                            "type": "string"
                        },
                        "TheThing": {
                            "$ref": "#/BugWorkaround"
                        }
                    }
                }
            }
        }
    }
}
```

**Comment by dblanchette** [comment]
Thanks for the detailed report. This should be fixed as of version 0.39.3 (just published). Feel free to reopen if not.

Happy holidays!
